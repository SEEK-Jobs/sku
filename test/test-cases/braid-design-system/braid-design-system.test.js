const path = require('path');
const { promisify } = require('util');
const rimrafAsync = promisify(require('rimraf'));
const fs = require('fs-extra');
const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const appDir = path.resolve(__dirname, 'app');
const distDir = path.resolve(appDir, 'dist');
const { getPathFromCwd } = require('../../../lib/cwd');

async function createPackageLink(name) {
  await fs.mkdirp(`${__dirname}/app/node_modules`);
  await fs.symlink(
    getPathFromCwd(`node_modules/${name}`),
    `${__dirname}/app/node_modules/${name}`,
  );
}

async function createPackageCopy(name) {
  const srcPath = getPathFromCwd(
    /^sku\//.test(name)
      ? `${name.replace('sku/', '')}`
      : `node_modules/${name}`,
  );
  const destPath = `${__dirname}/app/node_modules/${name}`;

  await fs.mkdirp(destPath);
  await fs.copy(srcPath, destPath);
}

async function copyLocalDependencies() {
  const nodeModules = `${__dirname}/app/node_modules`;
  await rimrafAsync(nodeModules);
  await Promise.all(['react', 'react-dom'].map(createPackageLink));
  await Promise.all(
    ['braid-design-system', 'sku/treat', 'sku/@loadable/component'].map(
      createPackageCopy,
    ),
  );
}

describe('braid-design-system', () => {
  beforeAll(async () => {
    // "Install" React and braid-design-system into this test app so that webpack-node-externals
    // treats them correctly.
    await copyLocalDependencies();
    await runSkuScriptInDir('build', appDir);
  });

  it('should generate the expected files', async () => {
    const files = await dirContentsToObject(distDir);
    expect(files).toMatchSnapshot();
  });

  it('should handle braid-design-system in tests', async () => {
    const { childProcess } = await runSkuScriptInDir('test', appDir);
    expect(childProcess.exitCode).toEqual(0);
  });
});
