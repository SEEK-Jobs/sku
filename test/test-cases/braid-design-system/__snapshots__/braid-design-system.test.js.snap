// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`braid-design-system should generate the expected files 1`] = `
Object {
  "externals.json": "[
  \\"react\\",
  \\"react-dom/server\\"
]",
  "index.html": "
  <!DOCTYPE html>
  <html>
    <head>
      <meta charset=\\"UTF-8\\">
      <title>My Awesome Project</title>
      <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/style.css\\" />
    </head>
    <body>
      <div id=\\"app\\"><div class=\\"_3lc311G _2dPpu8E _30kRq5M _3k5ze2u _2oY7eBs\\"><span class=\\"_3lc311G _1mZMC-p mT50rPc _2tWVdVE _26ZvMQL _28lA_OY\\">Hello World</span></div></div>
      <script type=\\"text/javascript\\" src=\\"/main.js\\"></script>
    </body>
  </html>
",
  "main.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "style.css": "._1-w6rIh,._1_nUkyU,._1AY55lW,._1Edrobi,._1eL7GTN,._1eMKFSn,._1FpPa02,._1g1DJL0,._1IhnU-p,._1lm-l5H,._1MGtKSw,._1MqFTwl,._1RhvqbX,._1tV8u3A,._1U3x5ln,._1v16zEs,._1WDu6Wh,._1yRM2wC,._2_WihXZ,._2AN0c48,._2fO5UlH,._2fYF4lo,._2g0nR3q,._2ibAU8h,._2ixMIHY,._2KgtkWp,._2Lwah7w,._2ojmSRG,._2OnDc7W,._2oTlOI9,._2oY7eBs,._2sDccJx,._2TjyCgn,._2TOt0dl,._2trBxvL,._2v89aJ_,._2wa5brK,._2WCNw0q,._3-Bbmhp,._3BH_6j9,._3BI4WO0,._3CmJ7Kl,._3Djm44j,._3DOO473,._3f95heZ,._3fBDMQ0,._3FK76Tr,._3gCqJVq,._3ICvoBB,._3IiUysg,._3LAqmg3,._3llRwqo,._3q6Ds3F,._3qmSVIt,._3RcCdvw,._3RXLvKp,._3SiKRpv,._3thgM40,._3vEQfRp,._3XJcmfN,._3xWreXo,._4diPrFH,._4P8f6gO,._9hFgAfn,._11NzP02,._15efk9s,._23DwT9L,._26bhdaL,._30uCamc,._33oqQo2,._112OfWK,._-7QbC7s,.AxDdU_m,.dKpIASS,.DoPP1qq,.gwBaSw8,.hmMX6s5,.IulIPMv,.JjROhPr,.mfwkjSB,.Q3O5KmD{margin:0;padding:0;border:0;font-size:100%;font:inherit;vertical-align:baseline}._1AY55lW,._1Edrobi,._1eL7GTN,._1yRM2wC,._2Lwah7w,._2oTlOI9,._2sDccJx,._2trBxvL,._3BI4WO0,._3xWreXo,._11NzP02{display:block}._2ojmSRG{line-height:1}._4diPrFH,.hmMX6s5{list-style:none}._2_WihXZ,._23DwT9L{quotes:none}._2_WihXZ:after,._2_WihXZ:before,._23DwT9L:after,._23DwT9L:before{content:\\"\\"}._1v16zEs{border-collapse:collapse;border-spacing:0}.DoPP1qq{background-color:transparent;color:inherit}
._2NXkmA_{padding-top:12px;padding-bottom:12px}._2NXkmA_,._26ZvMQL{font-size:16px;line-height:24px;-webkit-transform:translateY(.41em);transform:translateY(.41em)}._24XqWRc{font-size:18px;line-height:24px;-webkit-transform:translateY(.32666667em);transform:translateY(.32666667em)}
.othudig{background:#fff}._3YJXejn{background:#eee}._3vrXpN1{background:#2765cf}._23jaKqA{background:#ccc}.a_ARNp7{background:#f5f8ff}._1L54paI{background:#9556b7}._19O_nw1,.XKPC3x2{background:#fff}
._3k5ze2u{border-color:#d6d6d6}._9Ia9Rvl{border-color:#2765cf}._20dXyP-{border-color:#e60278}
._3Zn-PeU{border-radius:4px}
._1sseFqJ{border-style:solid;border-width:1px}
._1d6RKtE{-webkit-box-shadow:0 0 0 1px #2765cf;box-shadow:0 0 0 1px #2765cf}
.mT50rPc{color:#1c1c1c}._1Rsg9w6{color:#fff}._3OSCp5v{color:#e60278}.AsBZy5a{color:green}._1DYv5gu{color:#1c1c1ca1}._3rwI692{color:#2765cf}
._16w04Xz{fill:currentColor}._1srDitw{fill:#2765cf}.hK-SldW{fill:#ccc}._3SkkQl1{fill:#e60278}.q2J3ENx{fill:green}._2WNYdGY{fill:#1c1c1ca1}.QR_XQhw{fill:#fff}
._2tWVdVE{font-family:Helvetica Neue,HelveticaNeue,Helvetica,Arial,sans-serif}
._28lA_OY{font-weight:400}.OFyAM3C{font-weight:600}
._2XfQI3Q{margin-top:0}._1L7e8Uc{margin-top:6px}._2DeEfmd{margin-top:12px}._2BrS2wh{margin-top:18px}._87KrWkV{margin-top:24px}._31oD2hw{margin-top:30px}._3G8Ovpy{margin-top:48px}._2uO0VaK{margin-top:72px}
._2MHtdsp{margin-right:0}._1zC4Aqv{margin-right:25px}._3PgwENl{margin-right:5px}.hHgpRKb{margin-right:10px}._1lGUeYP{margin-right:15px}._20gfRgU{margin-right:20px}._3shmQ7C{margin-right:25px}._2Rtm0Y3{margin-right:40px}._2UBbbh2{margin-right:60px}
._3BWOXEL{margin-bottom:0}._1ITeVTC{margin-bottom:6px}.N17WSb0{margin-bottom:12px}._22treAZ{margin-bottom:18px}._1hO7oa0{margin-bottom:24px}._2iPPJdL{margin-bottom:30px}._7wzGp-O{margin-bottom:48px}._3o9aUxR{margin-bottom:72px}
._12PkTa8{margin-left:0}._2FpeYqQ{margin-left:25px}.B641TXX{margin-left:5px}.LbTXLuL{margin-left:10px}._2pdrpSr{margin-left:15px}._3bx9d4t{margin-left:20px}._1WdNA2z{margin-left:25px}._1ygksYE{margin-left:40px}.rUArqh6{margin-left:60px}
._3NMOU2X{padding-top:0}._1ls-jZ1{padding-top:6px}.lkpjUVN{padding-top:12px}._2gaj5e8{padding-top:18px}._3r2ZzV1{padding-top:24px}.j8YPtI7{padding-top:30px}._2f3XCMb{padding-top:48px}._2OwFoZL{padding-top:72px}
._2bajePk{padding-right:0}.l7bklSq{padding-right:25px}._3nGUVs9{padding-right:5px}._3IXJ1ic{padding-right:10px}._1HY9cSz{padding-right:15px}._2JtfYKn{padding-right:20px}.N_IBLVM{padding-right:25px}._2rAxYeA{padding-right:40px}._38ivZMt{padding-right:60px}
._12ovpwW{padding-bottom:0}._35vhuqv{padding-bottom:6px}._35G7cN1{padding-bottom:12px}._3oC_jYN{padding-bottom:18px}._1cn_feP{padding-bottom:24px}._3yAttcJ{padding-bottom:30px}._2C262aU{padding-bottom:48px}.JM2NZMA{padding-bottom:72px}
._2oI2UPd{padding-left:0}._12qnJF8{padding-left:25px}._2DMw3tD{padding-left:5px}.C6wfDbl{padding-left:10px}._2qI5mDP{padding-left:15px}._1cTIDw3{padding-left:20px}._3kQGh81{padding-left:25px}._1GiKpCt{padding-left:40px}._2lM6EAP{padding-left:60px}
._3MpHx96{-webkit-transition:all .125s ease;transition:all .125s ease}
._2dPpu8E{-webkit-box-sizing:border-box;box-sizing:border-box}
._3lc311G{display:block}._1CaVPtc{display:list-item;list-style:disc}
",
}
`;
